### PR TITLE
Use libetpan API for fetching whole IMAP attachment to file

### PR DESCRIPTION
**Pull request by disaykin**

### CODE REVIEW

File: src/core/imap/MCIMAPSession.h [hunk 1]
```diff
@@ -315,6 +320,7 @@ namespace mailcore {
         Data * fetchNonDecodedMessageAttachment(String * folder, bool identifier_is_uid,
                                       uint32_t identifier, String * partID,
                                       bool wholePart, uint32_t offset, uint32_t length,
+                                      IMAPMsgBodyHandler msg_body_handler, void * msg_body_handler_context,
```
Comment (dinhviethoa) [inline src/core/imap/MCIMAPSession.h]: Could you declare the function inline here instead?

**Comment by disaykin** [comment]
Sorry, but I don't understand what you mean.

File: src/core/imap/MCIMAPSession.h [hunk 2]
```diff
@@ -123,6 +123,9 @@ namespace mailcore {
                                                    Encoding encoding, IMAPProgressCallback * progressCallback, ErrorCode * pError);
 
         virtual void fetchMessageAttachmentToFileByUID(String * folder, uint32_t uid, String * partID,
+                                                       Encoding encoding, String * outputFile,
+                                                       IMAPProgressCallback * progressCallback, ErrorCode * pError);
+        virtual void fetchMessageAttachmentToFileByUID(String * folder, uint32_t uid, String * partID,
```
Comment (dinhviethoa) [inline src/core/imap/MCIMAPSession.h]: Last change: can you rename `fetchMessageAttachmentToFileByUID` to `fetchMessageAttachmentToFileByChunksByUID`.

**Comment by dinhviethoa** [comment]
Everything looks good now (one last minor change)!
Thank you so much for that work.

**Comment by disaykin** [comment]
Unfortunately, this version crashed on `mailimap_set_msg_body_handler(mImap, msg_body_handler, decoder);`.

**Comment by dinhviethoa** [comment]
I guess you have a fix in mind already 🎱

**Comment by disaykin** [comment]
I have a couple of ideas. But they are not perfect.

**Comment by dinhviethoa** [comment]
Could you show the crash you're running into?

**Comment by disaykin** [comment]
Crash occured because mImap == NULL on fresh connections. The method fetchNonDecodedMessageAttachment does not crash because it calls selectIfNeeded() before doing their work. We need call setup(), or connectIfNeeded() or selectIfNeeded() before setting up msg_body_handler.

**Comment by dinhviethoa** [comment]
an additional `selectIfNeeded(folder, pError)` in fetchMessageAttachmentToFileByUID and fetchMessageAttachmentToFileByChunksByUID won't hurt.

**Comment by disaykin** [comment]
Great!

**Comment by disaykin** [comment]
Fixed

**Pull request merged**